File: app-android/app/src/main/java/de/tutao/tutanota/data/KeyValueDao.java
package de.tutao.tutanota.data;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.room.Dao;
import androidx.room.Query;

@Dao
public interface KeyValueDao {
	@Query("SELECT value FROM KeyValue WHERE key = :key LIMIT 1")
	@Nullable
	String getString(String key);

	@Query("INSERT OR REPLACE INTO KeyValue (key, value) VALUES (:key, :value)")
	void putString(@NonNull String key, @Nullable String value);

	@Query("SELECT CAST(value as INT) FROM KeyValue WHERE key = :key LIMIT 1")
	long getLong(String key);

	@Query("INSERT OR REPLACE INTO KeyValue (key, value) VALUES (:key, :value)")
	void putLong(@NonNull String key, long value);
}

File: app-android/tutashared/src/main/java/de/tutao/tutashared/data/KeyValueDao.kt
package de.tutao.tutashared.data

import androidx.room.Dao
import androidx.room.Query

@Dao
interface KeyValueDao {
	@Query("SELECT value FROM KeyValue WHERE key = :key LIMIT 1")
	fun getString(key: String): String?

	@Query("INSERT OR REPLACE INTO KeyValue (key, value) VALUES (:key, :value)")
	fun putString(key: String, value: String?)

	@Query("SELECT CAST(value as INT) FROM KeyValue WHERE key = :key LIMIT 1")
	fun getLong(key: String): Long

	@Query("INSERT OR REPLACE INTO KeyValue (key, value) VALUES (:key, :value)")
	fun putLong(key: String, value: Long)
}